lib/mongoid/fields/validators/macro.rb

```diff
@@ -118,8 +115,9 @@ def validate_options(klass, name, options)
 
             if option == :type && options[option] == Symbol
               @field_type_is_symbol_warned ||= begin
-                Mongoid.logger.warn(FIELD_TYPE_IS_SYMBOL)
-                true
+                Mongoid::Deprecation.warn(<<~MSG.squish.freeze) && true
```

Comment:
@johnnyshields Could you explain the purpose of the freeze call here? Previously the message was stored in a constant and that constant's value was frozen. Now the message is created and consumed locally, what is the purpose of freezing it?

Reply:
Any strings which will not be later modified should be frozen.